pkg/3scale/amp/operator/upgrade.go

```diff
@@ -523,6 +531,43 @@ func ensureBackendRouteEnvVar(desired v1.EnvVar, existingEnvVars *[]v1.EnvVar) b
 	return update
 }
 
+func (u *UpgradeApiManager) upgradeZyncPodTemplateAnnotations() (reconcile.Result, error) {
+	zync, err := Zync(u.apiManager, u.Client())
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	desired := zync.DeploymentConfig()
+	existing := &appsv1.DeploymentConfig{}
+	err = u.Client().Get(context.TODO(), types.NamespacedName{Name: desired.Name, Namespace: u.apiManager.Namespace}, existing)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if existing.Annotations == nil {
+		existing.Annotations = map[string]string{}
+	}
+	update := false
+
+	for desiredAnnotationKey, desiredAnnotationVal := range desired.Annotations {
+		existingAnnotationVal, ok := existing.Annotations[desiredAnnotationKey]
```

Comment:
1.- Existing "wrong" annotations will no be deleted from existing `metadata.annotations`
2.- Desired annotations will not be included in `spec.template.metadata.annotations`

Reply:
this is still WIP (I had to test it yet) but you are right about 2 :+1: . It should be fixed now (pending testing). Thanks!

Reply:
About 1... I hoped our reconciliation of the metadata annotations might fix it (/home/msoriano/go/src/github.com/3scale/3scale-operator/pkg/helper/object_meta_merger.go) but I think it'd only merge desired and existing so it would not remove the old ones. I'll take a look at it when testing it :+1:

Reply:
About 1, Our object metamerger would not remove the old ones indeed.
I've updated the upgrade procedure to delete the old ones from `metadata.Annotations`.

Pending testing it/trying it

Reply:
It seems to work ok